### PR TITLE
Fix replication group e2e script, custom update, and use setStatusDefaults

### PR DESCRIPTION
**Pull request by kumargauravsharma**
Issue #404 

Description of changes:
* Fixed ReplicationGroup e2e test script
* Updated ReplicationGroup e2e test script wait logic
* Updated `replication_group/custom_update_api.go` to latest set output code.
* Added `TemporaryOutOfSync` error
* Added `setStatusDefaults` method and updated `sdk template` with it
* Updated sdkUpdate() call to invoke `setStatusDefaults()`
* Regenerated service controllers as sdk template got updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/pkg/crd_sdk.go.tpl [hunk 1]
```diff
@@ -375,3 +353,19 @@ func (rm *resourceManager) newDeleteRequestPayload(
 	return res, nil
 }
 {{- end -}}
+
+{{ "" }}
```
Comment (jaypipes) [inline templates/pkg/crd_sdk.go.tpl:357]: what's this?

File: test/e2e/elasticache/replication_group/e2e.sh [hunk 2]
```diff
@@ -35,8 +35,8 @@ setup_replication_group_fields
 wait_and_assert_replication_group_available_status() {
   sleep 5
   aws_wait_replication_group_available "$rg_id" "FAIL: expected replication group $rg_id to have been created in ${service_name}"
-  k8s_assert_replication_group_status_property "$rg_id" ".status" "available"
-  sleep 5
+  aws_assert_replication_group_status "$rg_id" "available"
+  sleep 35
```
Comment (jaypipes) [inline test/e2e/elasticache/replication_group/e2e.sh:39]: :+1:

**Review by jaypipes — APPROVED** [review]
I think there might be a copy-pasta thing introduced in the template:

```
{{ "" }}
```

but feel free to remove in a future PR. This is good stuff.

File: templates/pkg/crd_sdk.go.tpl (hunk 1 above)
Comment (kumargauravsharma) [reply templates/pkg/crd_sdk.go.tpl:357]: without this the generated .go files were not having newline between previous line `}` and `// setStatusDefaults ...` code that followed it.

Comment (jaypipes) [reply templates/pkg/crd_sdk.go.tpl:357]: Ah, OK. You can modify the behaviour of the newline-trimming of the Go template by modifying the code directly above here from this:

```
{{- end -}}
```

to this:

```
{{- end }}
```

See here for a little more information on these "trim markers":

https://golang.org/pkg/text/template/#hdr-Text_and_spaces

**Pull request merged**